app/renderer/components/Inspector/HighlighterRects.js

```diff
@@ -1,20 +1,152 @@
+
+      //Create new element obj which will be a +/- centroid
+      if (elementsByOverlap[key].length > 1) {
+        element = {
+          type: RENDER.EXPAND,
+          element: null,
+          parent: null,
+          properties: {
+            left: null,
+            top: null,
+            width: null,
+            height: null,
+            centerX: element.properties.centerX,
+            centerY: element.properties.centerY,
+            path: key,
+            keyCode: key,
+            container: null,
+            accessible: null
+          }
+        };
+        elements = elements.concat(this.newCoordinates(elementsByOverlap[key], key));
+      }
+      elements.push(element);
+    }
+
+    // Remove first element as it is an XML document node
+    elements.shift();
+
+    return elements;
+  }
+
+  // Create a new object for each element and determine its properties
+  findElementProps (element, prevElement, elements, overlaps) {
+    if (!element) {
+      return {};
+    }
+    const { scaleRatio, containerEl } = this.props;
+    const {x1, y1, x2, y2} = parseCoordinates(element);
+    const xOffset = this.highlighterXOffset || 0;
+    const posOffset = 0.03 * containerEl.offsetWidth;
+    const obj = {
+      type: RENDER.CENTROID,
+      element,
+      parent: prevElement,
+      properties: {
+        left: x1 / scaleRatio + xOffset || 0,
+        top: y1 / scaleRatio || 0,
+        width: (x2 - x1) / scaleRatio || 0,
+        height: (y2 - y1) / scaleRatio || 0,
+        centerX: ((Math.round(x1 + ((x2 - x1) / 2)) / scaleRatio) + xOffset - posOffset) || 0,
+        centerY: ((Math.round(y1 + ((y2 - y1) / 2)) / scaleRatio) - posOffset) || 0,
+        path: element.path,
+        keyCode: null,
+        container: false,
+        accessible: element.attributes.accessible
+      }
+    };
+    const coordinateName = `${obj.properties.centerX}.${obj.properties.centerY}`;
+    obj.properties.container = this.isElementContainer(obj, elements);
+    elements.push(obj);
+
+    overlaps[coordinateName] ?
+      overlaps[coordinateName].push(obj)
+      :
+      overlaps[coordinateName] = [obj];
+
+    if (element.children) {
+      for (let childEl of element.children) {
+        this.findElementProps(childEl, element, elements, overlaps);
+      }
+    }
+
+    return overlaps;
+  }
+
+  isElementContainer (element1, elements) {
+    for (let element2 of elements) {
+      if (element2.element !== element1.element
+        && this.isElementOverElement(element1.properties, element2.properties)
+        && !this.isAncestor(element1.parent, element2.element, elements)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  isElementOverElement (element1, element2) {
+    return element1.left <= element2.left
+        && element2.width <= element1.width
+        && element1.top >= element2.top
+        && element2.height <= element1.height;
+  }
+
+  isAncestor (parent, checkElement, elements) {
```

Comment:
it's a bit unclear what we're checking is an ancestor of what? is this function signature basically saying, "tell me if `parent` is an ancestor of `checkElement`"? or the other way around?